File: src/java/magma/app/compile/text/Placeholder.java
package magma.app.compile.text;

import magma.api.Type;
import magma.api.option.None;
import magma.api.option.Option;
import magma.app.Main;
import magma.app.compile.define.Definition;
import magma.app.compile.define.Parameter;
import magma.app.compile.type.PrimitiveType;
import magma.app.compile.value.Value;
import magma.app.io.Platform;

public record Placeholder(String input) implements Parameter, Value, Type {
    @Override
    public String generate() {
        return Main.generatePlaceholder(this.input);
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public String generate(final Platform platform) {
        return this.generate();
    }

    @Override
    public Option<Value> findChild() {
        return new None<Value>();
    }

    @Override
    public Option<Definition> asDefinition() {
        return new None<Definition>();
    }

    @Override
    public Option<Value> toValue() {
        return new None<Value>();
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }

    @Override
    public Option<String> generateAsEnumValue(final String structureName, Platform platform) {
        return new None<String>();
    }

    @Override
    public Type type() {
        return PrimitiveType.Auto;
    }
}

File: src/node/magma/app/compile/text/Placeholder.ts
/*[
	Actual, 
	Characters, 
	Collector, 
	CompileState, 
	Console, 
	Definition, 
	DivideState, 
	EmptyHead, 
	Err, 
	Files, 
	FlatMapHead, 
	FunctionHeader, 
	FunctionSegment, 
	Head, 
	HeadedQuery, 
	IOError, 
	ImmutableCompileState, 
	Import, 
	Joiner, 
	List, 
	ListCollector, 
	Lists, 
	MapHead, 
	Namespace, 
	None, 
	Ok, 
	Option, 
	Parameter, 
	Path, 
	Placeholder, 
	Queries, 
	Query, 
	RangeHead, 
	Result, 
	SingleHead, 
	Some, 
	Strings, 
	Tuple2, 
	Tuple2Impl, 
	Type, 
	ZipHead
]*/
import { Main } from "../../../../magma/app/Main";
import { Platform } from "../../../../magma/app/io/Platform";
import { Value } from "../../../../magma/app/compile/value/Value";
import { Option } from "../../../../magma/api/option/Option";
import { None } from "../../../../magma/api/option/None";
import { Definition } from "../../../../magma/app/compile/define/Definition";
import { Type } from "../../../../magma/api/Type";
import { PrimitiveType } from "../../../../magma/app/compile/type/PrimitiveType";
export class Placeholder {
	input: string;
	constructor (input: string) {
		this.input = input;
	}
	generate(): string {
		return Main/*auto*/.generatePlaceholder(this/*auto*/.input);
	}
	isFunctional(): boolean {
		return false/*auto*/;
	}
	generate(platform: Platform): string {
		return this/*auto*/.generate(/*auto*/);
	}
	findChild(): Option<Value> {
		return new None<Value>(/*auto*/);
	}
	asDefinition(): Option<Definition> {
		return new None<Definition>(/*auto*/);
	}
	toValue(): Option<Value> {
		return new None<Value>(/*auto*/);
	}
	isVar(): boolean {
		return false/*auto*/;
	}
	generateBeforeName(): string {
		return "";
	}
	generateAsEnumValue(structureName: string, platform: Platform): Option<string> {
		return new None<string>(/*auto*/);
	}
	type(): Type {
		return PrimitiveType/*auto*/.Auto;
	}
}
